java-checks/src/main/java/org/sonar/java/checks/security/ControllingPermissionsCheck.java

```diff
@@ -0,0 +1,141 @@
+    "org.springframework.security.access.expression.method.MethodSecurityExpressionHandler",
+    "org.springframework.security.core.GrantedAuthority",
+    "org.springframework.security.acls.model.PermissionGrantingStrategy"));
+
+  private static final List<String> CLASSES = Collections.singletonList("org.springframework.security.config.annotation.method.configuration.GlobalMethodSecurityConfiguration");
+
+  private static final String MUTABLE_ACL_SERVICE = "org.springframework.security.acls.model.MutableAclService";
+  private static final MethodMatcherCollection METHOD_MATCHERS = MethodMatcherCollection.create(
+    MethodMatcher.create().typeDefinition(MUTABLE_ACL_SERVICE).name("createAcl").withAnyParameters(),
+    MethodMatcher.create().typeDefinition(MUTABLE_ACL_SERVICE).name("deleteAcl").withAnyParameters(),
+    MethodMatcher.create().typeDefinition(MUTABLE_ACL_SERVICE).name("updateAcl").withAnyParameters(),
+    MethodMatcher.create().typeDefinition("org.springframework.security.config.annotation.web.builders.HttpSecurity").name("authorizeRequests").withAnyParameters());
+
+  private JavaFileScannerContext context;
+
+  @Override
+  public void scanFile(JavaFileScannerContext context) {
+    if (context.getSemanticModel() == null) {
+      return;
+    }
+    this.context = context;
+    scan(context.getTree());
+  }
+
+  @Override
+  public void visitClass(ClassTree tree) {
+    if (!tree.symbol().isInterface()) {
+      tree.superInterfaces().forEach(superInterface -> {
+        Type interfaceType = superInterface.symbolType();
+        if (INTERFACES.stream().anyMatch(interfaceType::is)) {
+          context.reportIssue(this, superInterface, MESSAGE);
+        }
+      });
+      TypeTree superClass = tree.superClass();
+      if (superClass != null && CLASSES.stream().anyMatch(superClass.symbolType()::is)) {
+        context.reportIssue(this, superClass, MESSAGE);
+      }
+    }
+    tree.modifiers().annotations().forEach(annotationTree -> {
+      Type annotationType = annotationTree.symbolType();
+      if (JSR_250_ANNOTATIONS.stream().anyMatch(annotationType::is)) {
+        context.reportIssue(this, annotationTree, MESSAGE);
+      }
+    });
+    super.visitClass(tree);
+  }
+
+  @Override
+  public void visitNewClass(NewClassTree tree) {
+    if (tree.classBody() != null) {
+      Type newClassType = tree.symbolType();
+      // only target anonymous classes
+      if (INTERFACES.stream().anyMatch(newClassType::isSubtypeOf)
+        || CLASSES.stream().anyMatch(newClassType::isSubtypeOf)) {
+        context.reportIssue(this, tree.newKeyword(), tree.arguments(), MESSAGE);
+      }
+    }
+    super.visitNewClass(tree);
+  }
+
+  @Override
+  public void visitMethod(MethodTree tree) {
+    tree.modifiers().annotations().forEach(annotationTree -> {
+      Type annotationType = annotationTree.symbolType();
+      if (Stream.concat(ANNOTATIONS.stream(), JSR_250_ANNOTATIONS.stream()).anyMatch(annotationType::is)) {
```

Comment:
I would just go for a simple || here, but this is mainly styling matter, I find the concat confusing.